### PR TITLE
Metadata collection refactoring

### PR DESCRIPTION
**Pull request by masci**
### What does this PR do?

This is a rewrite of the Metadata collection, it was somehow on the todo list given the naive approach we used for the first draft.

This PR introduces the concept of `Metadata Collector` along with the existing `Metadata Provider`.

### Motivation

The agent codebase is going to be used in different environments other than the Agent itself and this applies also to Metadata Providers (see #270). To be able to import a Metadata Provider as a regular Go package from different projects we need to keep it loosely coupled to the rest of the agent in order to minimize dependencies. The drawback is we need something else responsible to actually send the metadata payload to the intake, thus the need of `Metadata Collectors`. 

**TL;DR** description of the requirements:

 * A Metadata Provider is responsible to define a payload and gather the informations needed to fill it (this is non-agent specific).
 * A Metadata Collector is responsible to periodically invoke one or more Providers, build and send the payload to the proper endpoint in the intake API.

### Additional Notes

This is a first step, once merged I'll break down the `host` metadata package so it can be used without an embedded CPython interpreter (this is needed to avoid bloating Dogstatsd, where we need host metadata without the Python info).

### CODE REVIEW

**Review by olivielpeau — APPROVED** [review]
A few nits, but overall this looks great! ⭐️  (nice docs too)

**Review by olivielpeau — APPROVED** [review]
one comment on the naming convention, but LGTM otherwise

**Review by olivielpeau — APPROVED** [review]
